src/test/java/uk/gov/hmcts/darts/audio/service/impl/AudioServiceImplTest.java

```diff
@@ -218,6 +230,48 @@ void addAudio() throws IOException {
         assertEquals("1", savedMedia.getCourtroom().getName());
     }
 
+    @Test
+    void handheldAudioShouldNotLinkAudioToHearingByEvent() {
+
+        AddAudioMetadataRequest addAudioMetadataRequest = createAddAudioRequest(STARTED_AT, ENDED_AT);
+        addAudioMetadataRequest.setTotalChannels(1);
+
+        HearingEntity hearing = new HearingEntity();
+        EventEntity eventEntity = new EventEntity();
+        eventEntity.addHearing(hearing);
+
+        MediaEntity mediaEntity = createMediaEntity(STARTED_AT, ENDED_AT);
+
+        when(audioConfigurationProperties.getHandheldAudioCourtroomNumbers())
+            .thenReturn(Arrays.asList(addAudioMetadataRequest.getCourtroom()));
+
+        audioService.linkAudioToHearingByEvent(addAudioMetadataRequest, mediaEntity);
+        verify(hearingRepository, times(0)).saveAndFlush(any());
+        assertEquals(0, hearing.getMediaList().size());
+    }
+
+    @Test
+    void linkAudioToHearingByEvent() {
```

Comment:
does this test cover pre-amble only? is it worth having a test for post-amble / both?

Reply:
The response from searching for events is mocked so a post-amble test would in theory be a duplicate. I have however added an additional test to cover when the query brings back more than one court log for the same hearing to check that the audio is only linked once - and have used post amble times for the court log events.